src/modal.ts

```diff
@@ -0,0 +1,462 @@
+    DEFAULT_VALUES:{ [key:string]: any } = {
+        pos:     "xs:bottom;m:top-center",
+        height:  "xs:50vh;m:auto",
+        width:   "xs:100%;m:50vw",
+        animation: "xs:b2t",
+        position_breakpoints : [{cond:"(min-width:0px) and (max-width:992px)",
+                                 val:"bottom"},
+                                {cond: "(min-width:992px)", val: "top-center"}],
+        height_breakpoints : [{cond: "(min-width:0px) and (max-width:992px)",
+                               val: "50vh"},
+                             {cond: "(min-width:992px)", val: "auto"}],
+        width_breakpoints : [{cond: "(min-width:0px) and (max-width:992px)",
+                              val: "100%"},
+                             {cond: "(min-width:992px)", val: "50vw"}],
+        animate_breakpoints : [{cond: "(min-width:0px)", val: "b2t",
+                                ul:null, ll:0}]
+    }
+    @property({type:Boolean, reflect:true, attribute:true})
+    open = false;
+
+    @property({type:Boolean, attribute:true})
+    noesc = false;
+
+    @property({type:Boolean, attribute:true})
+    nobgclose = false;
+
+    @property({type:Boolean, attribute:true})
+    stack = false;
+
+    @property({type:String, attribute:true})
+    guidesel = null;
+
+    @property({type:String, attribute:true})
+    pos = this.DEFAULT_VALUES.pos;
+
+    @property({type:String, attribute:true})
+    height = this.DEFAULT_VALUES.height;
+
+    @property({type:String, attribute:true})
+    width = this.DEFAULT_VALUES.width;
+
+    @property({type:String, attribute:true})
+    animation = this.DEFAULT_VALUES.animation;
+
+    @property({type:String, attribute:true})
+    closesel = null;
+
+    @state()
+    animationDur = .3;
+
+    @state()
+    _position_breakpoints = this.DEFAULT_VALUES.position_breakpoints;
+
+    @state()
+    _height_breakpoints = this.DEFAULT_VALUES.height_breakpoints;
+
+    @state()
+    _width_breakpoints = this.DEFAULT_VALUES.width_breakpoints;
+
+    @state()
+    _animate_breakpoints = this.DEFAULT_VALUES.animate_breakpoints;
+
+    @state()
+    _guide_value:{ [key:string]:any} = {};
+
+    @state()
+    _open = false;
+
+    @state()
+    _mouse_pos_mod = null;
+
+    @state()
+    _change_mc = false;
+
+    @state()
+    timer_obj:ReturnType<typeof setTimeout>|null = null;
+
+    @query('#__wrapper')
+    _wrapper_el:HTMLElement;
+
+    actualParent:HTMLElement|null;
+
+    private _mouse_up_listener:any;
+    private _mouse_move_listener:any;
+    private _mouse_prev_x:number;
+    private _mouse_prev_y:number;
+    private _wrp_init_bounds:any;
+
+    override attributeChangedCallback(name:string, old_val:string,
+                                      new_val:string) {
+        super.attributeChangedCallback(name, old_val, new_val);
+        if (name == 'open') {
+            requestAnimationFrame(() => {
+             this.open?this.show():this.hide()
+            });
+        }
+        if (name == 'height'){
```

Comment:
This can very well be `else if`, if you are keeping if in all the cases, all the conditions will be unnecessarily checked. name will be one among these conditions only. Replace all the `if`s to `if else`s in this function.